File: java/Senjata.java
// Kelas induk: menyimpan data dasar senjata (5 atribut)
public class Senjata {

    private String idSenjata;   // ID unik senjata
    private String nama;        // Nama senjata
    private String jenis;       // Jenis (Pistol/Rifle/Shotgun/SMG)
    private String spesifikasi; // Detail kaliber & fitur
    private String madeIn;      // Negara pembuat

    // Constructor tanpa parameter
    public Senjata() {}

    // Constructor dengan parameter
    public Senjata(String idSenjata, String nama, String jenis, String spesifikasi, String madeIn) {
        this.idSenjata   = idSenjata;
        this.nama        = nama;
        this.jenis       = jenis;
        this.spesifikasi = spesifikasi;
        this.madeIn      = madeIn;
    }

    // Getter Method
    public String getIdSenjata()   { return idSenjata; }
    public String getNama()        { return nama; }
    public String getJenis()       { return jenis; }
    public String getSpesifikasi() { return spesifikasi; }
    public String getMadeIn()      { return madeIn; }

    // Setter Method
    public void setIdSenjata(String idSenjata)     { this.idSenjata = idSenjata; }
    public void setNama(String nama)               { this.nama = nama; }
    public void setJenis(String jenis)             { this.jenis = jenis; }
    public void setSpesifikasi(String spesifikasi) { this.spesifikasi = spesifikasi; }
    public void setMadeIn(String madeIn)           { this.madeIn = madeIn; }
}

File: cpp/Senjata.cpp
#include <bits/stdc++.h>
using namespace std;

// Kelas induk: menyimpan data dasar senjata (5 atribut)
class Senjata {

private:
    string idSenjata;   // ID unik senjata
    string nama;        // Nama senjata
    string jenis;       // Jenis (Pistol/Rifle/Shotgun/SMG)
    string spesifikasi; // Detail kaliber & fitur
    string madeIn;      // Negara pembuat

public:
    // Constructor kosong
    Senjata() {}

    // Constructor dengan parameter
    Senjata(string id, string n, string j, string s, string m) {
        idSenjata   = id;
        nama        = n;
        jenis       = j;
        spesifikasi = s;
        madeIn      = m;
    }

    // Getter Method
    string getIdSenjata()const{
        return idSenjata;
    }
    string getNama()const{
        return nama;
    }
    string getJenis()const{
        return jenis;
    }
    string getSpesifikasi()const{
        return spesifikasi;
    }
    string getMadeIn()const{
        return madeIn;
    }

    // Setter Method
    void setIdSenjata(string id){
        idSenjata = id;
    }
    void setNama(string n){
        nama = n;
    }
    void setJenis(string j){
        jenis = j;
    }
    void setSpesifikasi(string s){
        spesifikasi = s;
    }
    void setMadeIn(string m){
        madeIn = m;
    }

    // Destructor
    virtual ~Senjata() {}
};
